### DISCUSSION TITLE
TextComponents to console strings

### DISCUSSION THREAD

**Discussion by SwordOfSouls** [Minestom/Minestom#1255] [Help] [answered]
Hey! So I'm trying to print the players name to console on join but am getting the raw TextComponent "TextComponentImpl{content="Islappedmykey", style=StyleImpl{color=null, obfuscated=not_set, bold=not_set, strikethrough=not_set, underlined=not_set, italic=not_set, clickEvent=null, hoverEvent=null, insertion=null, font=null}, children=[]}" Is there a way to convert this Component into a simple string? Another thing is when using the .log func in the Logger no matter what the text shows up as red in Intellij. Any help is greatly appreciated, thanks

**Comment by FluxCapacitor2** [comment, marked as answer]
If I remember correctly you can use `PlainTextComponentSerializer.plainText().serialize(your component)` to convert the component into a regular string with no colors.

Also, if you are just trying to print the username you can use `player.username` instead of `player.name`/`player.displayName`
